### PR TITLE
[TorqueBalancing] Added option to load torque PID gains from config

### PR DESCRIPTION
**Pull request by francesco-romano**
It is now possible to add at configuration file the possibility to load PID gains for the low level torque control.
A first use is to load a single set of gains. In this case, if no other options are specified, this gains are used for the duration of the module. During module closure the original gains are applied.
This should allow to launch torque balancing with different set of gains depending on the application (single foot, yoga) thus solving the last obstacle to do torquebalancing on the new ethernet robots.

A different use case is the possibility to load different set of gains at startup. It is then possible to switch between gains by using a newly introduced RPC command.
Further debug is need before considering a run-time switch of the gains, so this feature should be considere EXPERIMENTAL

@traversaro  @DanielePucci

### CODE REVIEW

**Comment by iron76** [comment]
:+1:

File: src/modules/torqueBalancing/src/TorqueBalancingModule.cpp [hunk 1]
```diff
@@ -224,6 +281,16 @@ namespace codyco {
...
 
+            //check torque gains after init
+            m_defaultTorquePIDsKey = rf.check("defaultTorqueGainsKey", Value::getNullValue(), "Checking default PIDs set");
+            if (rf.check("torqueGains", "Looking for torque gains section")) {
+                if (!loadTorqueGains(rf.find("torqueGains"))) {
+                    yError("A torque gain option is specified, but failed to load");
+                    return false;
+                }
+            }
+            std::cerr << "Gains in memory: " << m_torquePIDs.size() << "\n";
```
Comment (traversaro) [inline src/modules/torqueBalancing/src/TorqueBalancingModule.cpp]: Spurious debug print?

File: src/modules/torqueBalancing/README.md [hunk 2]
```diff
@@ -45,6 +45,22 @@ Users willing to use the module should follow this list.
...
+
+It is possible to specify multiple set of torque gains in order to switch them at runtime (see RPC section).
+As an example:
+
+- Single set: `torqueGains torqueGains.ini`
+- Multiple sets: `torqueGains ((set1, torqueGains_first.ini), (set2, torqueGains_second.ini))`
+
+The file contains for each joint a list of values for the proportional, integrative and derivative gains, e.g.
+`l_shoulder_roll     = ("kp", 5.0), ("ki", 0.0),("kd", 4.0)`
+The original gains read from the firmware will be overwritten with the specified gains in the configuration file (it is then possible to specify only the gains that **must** change)
```
Comment (traversaro) [inline src/modules/torqueBalancing/README.md]: @francesco-romano  Loading the gains present in the firmware to restore them at the end of the module is involving to much work in your opinion?

Comment (francesco-romano) [inline src/modules/torqueBalancing/README.md]: @traversaro  I added in the documentation that gains will be restored.

**Pull request merged**